elements/lisk-p2p/test/unit/peer_pool.ts

```diff
@@ -71,14 +81,539 @@ describe('peerPool', () => {
+				.which.equals(peerPoolConfig.peerSelectionForRequest);
+		});
+
+		it('should set _peerSelectForConnection property', async () => {
+			expect(peerPool)
+				.to.have.property('_peerSelectForConnection')
+				.which.equals(peerPoolConfig.peerSelectionForConnection);
+		});
+
+		it('should set _maxOutboundConnections property', async () => {
+			expect(peerPool)
+				.to.have.property('_maxOutboundConnections')
+				.which.equals(peerPoolConfig.maxOutboundConnections);
+		});
+
+		it('should set _maxInboundConnections property', async () => {
+			expect(peerPool)
+				.to.have.property('_maxInboundConnections')
+				.which.equals(peerPoolConfig.maxInboundConnections);
+		});
+
+		it('should set _sendPeerLimit property', async () => {
+			expect(peerPool)
+				.to.have.property('_sendPeerLimit')
+				.which.equals(peerPoolConfig.sendPeerLimit);
+		});
+
+		it('should have a _outboundShuffleIntervalId property', async () => {
+			expect(peerPool).to.have.property('_outboundShuffleIntervalId');
+		});
+	});
+
+	describe('#applyNodeInfo', () => {
+		it('should set _nodeInfo', async () => {
+			peerPool.applyNodeInfo(nodeInfo);
+
+			expect(peerPool.nodeInfo).to.equal(nodeInfo);
+		});
+
+		it('should call getPeers', async () => {
+			const getPeersStub = sandbox.stub(peerPool, 'getPeers').callThrough();
+			peerPool.applyNodeInfo(nodeInfo);
+
+			expect(getPeersStub).to.be.calledOnce;
+		});
+
+		it('should call _applyNodeInfoOnPeer for each peer in peerMap', async () => {
+			const applyNodeInfoOnPeerStub = sandbox
+				.stub(peerPool as any, '_applyNodeInfoOnPeer')
+				.callThrough();
+			const applyNodeInfoOnPeerCalls = applyNodeInfoOnPeerStub.getCalls()
+				.length;
+
+			expect(applyNodeInfoOnPeerCalls).eql(peerPool.getPeers().length);
+		});
+	});
+
+	describe('#request', () => {
+		it('should call getUniqueOutboundConnectedPeers', async () => {
+			sandbox.stub(peerPool, 'requestFromPeer').resolves();
+			const getUniqueOutboundConnectedPeersStub = sandbox
+				.stub(peerPool, 'getUniqueOutboundConnectedPeers')
+				.returns([peerInfo]);
+			await peerPool.request(requestPacket);
+
+			expect(getUniqueOutboundConnectedPeersStub).to.be.calledOnce;
+		});
+
+		it('should call _peerSelectForRequest', async () => {
+			sandbox.stub(peerPool, 'requestFromPeer').resolves();
+			const peers = peerPool.getUniqueOutboundConnectedPeers();
+			const _peerSelectForRequestStub = sandbox
+				.stub(peerPool as any, '_peerSelectForRequest')
+				.returns([peerInfo]);
+
+			await peerPool.request(requestPacket);
+
+			expect(_peerSelectForRequestStub).to.be.calledWith({
+				peers,
+				nodeInfo: peerPool.nodeInfo,
+				peerLimit: 1,
+				requestPacket,
+			});
+		});
+
+		it('should throw error if no peers selected', async () => {
+			sandbox.stub(peerPool as any, '_peerSelectForRequest').returns([]);
+			try {
+				await peerPool.request(requestPacket);
+			} catch (err) {
+				expect(err).to.be.instanceOf(RequestFailError);
```

Comment:
this could be false positive if the request doesn't throw.
It's should have
```
let err;
try {
...
} catch (error) {
  err = error;
}
expect(err).to.be.instanceOf(RequestFailError);
```